heareval/tasks/pipeline.py

```diff
@@ -556,6 +557,119 @@ def run(self):
         self.mark_complete()
 
 
+class ResampleSubcorpuses(WorkTask):
+    """
+    Aggregates resampling of all the splits and sampling rates
+    into a single task as dependencies.
```

Comment:
Why? Won't this block and prevent parallelization? What's the benefit?

Reply:
The benefit is that 
- We don't need to write the list of resampling tasks in the requirements of all other tasks after the `ResampleSubcorpus` and before the `FinalizeCorpus`. Since the `GenerateAudioStats` was something in the middle, I had to write the resample list in the requirements of both the `GenerateAudioStats` and `FinalizeCorpus`. 

Resample list which is referred above - 
https://github.com/neuralaudio/hear-eval-kit/blob/64fe04f24e4cd754495a95c127957c6d6504448a/heareval/tasks/pipeline.py#L579-L589

- This is consistent with how we are doing the `SubsampleSplits` which aggregate the `SubsampleSplit`. This will not affect parallelism as it is just a reference to the list of resampling tasks that will run in parallel.
https://github.com/neuralaudio/hear-eval-kit/blob/8365c0d2eac8b94b9b56af6b422087b6e33eab5b/heareval/tasks/pipeline.py#L286-L318

Reply:
It won't block parallelization, it basically just moves all separate requires tasks from finalizeCorpus out into a single task and aggregates the results into a single requires workdir for the GenerateAudioStats which simplifies that task. Is that correct @khumairraj ?

Reply:
Yes.